### ISSUE TITLE
Fix docker publishing

### ISSUE THREAD

**Issue by MartinKolarik** [algolia/npm-search#1233]
I fixed an issue with the script path in #1232 but now publishing fails with:

```
SemanticReleaseError: Cannot push to the Git repository.
    at default (file:///home/runner/work/npm-search/npm-search/node_modules/semantic-release/lib/get-error.js:6:10)
    at run (file:///home/runner/work/npm-search/npm-search/node_modules/semantic-release/index.js:101:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async default (file:///home/runner/work/npm-search/npm-search/node_modules/semantic-release/index.js:275:22)
    at async file:///home/runner/work/npm-search/npm-search/scripts/publish-check.mjs:[19](https://github.com/algolia/npm-search/actions/runs/7248615388/job/19744902325#step:5:20):25 {
  code: 'EGITNOPERMISSION',
  details: '**semantic-release** cannot push the version tag to the branch `master` on the remote Git repository with URL `https://x-access-token:[secure]@github.com/algolia/npm-search.git`.\n' +
    '\n' +
    'This can be caused by:\n' +
    ' - a misconfiguration of the [repositoryUrl](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#repositoryurl) option\n' +
    ' - the repository being unavailable\n' +
    ' - or missing push permission for the user configured via the [Git credentials on your CI environment](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/ci-configuration.md#authentication)',
  semanticRelease: true
}
```

I believe this might be related to branch protection rules, but don't have high enough access to see those, so not sure. Also, the whole CI job ends without reporting errors even though it fails, which isn't great: https://github.com/algolia/npm-search/actions/runs/7248615388/job/19744902325

@pixelastic can you take a look please?

**Comment by pixelastic** [comment]
Thanks for noticing. I'll have a look shortly. 

Doesn't seem to be related to docker per-se, it's the branching pre-check that validates if we need to do a release or not. I don't understand why it would need to push to the repo, I'll investigate.

**Comment by MartinKolarik** [comment]
> I don't understand why it would need to push to the repo, I'll investigate.

For this I think: https://github.com/algolia/npm-search/commits?author=semantic-release-bot

**Comment by pixelastic** [comment]
Issue should now be fixed.

I made sure the workflow would now be marked either as `failed` (if error in the scripts), `success` (if everything went well), or `skipped` (if no release was necessary). I had to update some workflow permission (in the .yml file), so it could correctly cancel a run and write tags/releases.

I had to disable the "Require status checks to pass before merging" branch protection. I thought it would only apply on PRs (to prevent merging a PR where the `VALIDATE` step doesn't succeed), but it was also preventing `semantic-release` from adding a new tag. I'm not entirely sure why (I assume `semantic-release` was creating a branch and merging it to master but not `VALIDATE` was ever run on it, so GH would block it).

In practice it shouldn't have much of an impact: we already run `VALIDATE` as part of our publishing workflow, so we can be sure it will be run before publishing. It only means that we can now technically merge PRs, even if the `VALIDATE` check doesn't pass.
